src/extensions/scratch3_wedo2/index.js

```diff
@@ -1543,8 +1543,10 @@ class Scratch3WeDo2Blocks {
     _isTilted (direction) {
         switch (direction) {
         case WeDo2TiltDirection.ANY:
-            return (Math.abs(this._peripheral.tiltX) >= Scratch3WeDo2Blocks.TILT_THRESHOLD) ||
-                (Math.abs(this._peripheral.tiltY) >= Scratch3WeDo2Blocks.TILT_THRESHOLD);
+            return (this._peripheral.tiltX > 45 ? 256 - this._peripheral.tiltX : this._peripheral.tiltX) >=
```

Comment:
Can we make a named constant for what 45 and 256 are?
Also, can you add comments explaining the bug to this code?

Pulling `this._peripheral.tiltX > 45 ? 256` into a boolean variable might also make this more readable

Reply:
I agree with you, it's not easy-to-read. On second thought, as the two constants(45, 256) come from other function [_getTiltAngle()](https://github.com/LLK/scratch-vm/blob/7af161f1a2e4cadd02d747d77f6cea0299937581/src/extensions/scratch3_wedo2/index.js#L1559), I think it would be better the tilt value conversion is calculated in the single point where is the function _getTiltAngle(). So can I suggest changing these lines as follows so that the complicated conversion is not needed in _isTilted() anymore?

``` JavaScript
return this._getTiltAngle(WeDo2TiltDirection.UP) >= Scratch3WeDo2Blocks.TILT_THRESHOLD ||
    this._getTiltAngle(WeDo2TiltDirection.DOWN) >= Scratch3WeDo2Blocks.TILT_THRESHOLD ||
    this._getTiltAngle(WeDo2TiltDirection.LEFT) >= Scratch3WeDo2Blocks.TILT_THRESHOLD ||
    this._getTiltAngle(WeDo2TiltDirection.RIGHT) >= Scratch3WeDo2Blocks.TILT_THRESHOLD;

Reply:
Ah, I see, that function doesn't explain what those numbers are either. Yes, I think that would make a lot of sense!